x-pack/legacy/plugins/apm/server/lib/ui_filters/local_ui_filters/index.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Server } from 'hapi';
+import { cloneDeep, sortByOrder } from 'lodash';
+import { mergeProjection } from '../../../../public/projections/util/merge_projection';
+import { UIFilters } from '../../../../typings/ui-filters';
+import { Projection } from '../../../../public/projections/typings';
+import { PromiseReturnType } from '../../../../typings/common';
+import { getFilterAggregations } from './get_filter_aggregations';
+import { Setup } from '../../helpers/setup_request';
+import { localUIFilters, LocalUIFilterName } from './config';
+
+export type LocalUIFiltersAPIResponse = PromiseReturnType<
+  typeof getLocalUIFilters
+>;
+
+export async function getLocalUIFilters({
+  server,
+  setup,
+  projection,
+  uiFilters,
+  localFilterNames
+}: {
+  server: Server;
+  setup: Setup;
+  projection: Projection;
+  uiFilters: UIFilters;
+  localFilterNames: LocalUIFilterName[];
+}) {
+  const { client } = setup;
+
+  const projectionWithoutAggs = cloneDeep(projection);
+
+  delete projectionWithoutAggs.body.aggs;
+
+  const filterAggregations = await getFilterAggregations({
+    server,
+    uiFilters,
+    projection,
+    localFilterNames
+  });
+
+  const params = mergeProjection(projectionWithoutAggs, {
+    body: {
+      size: 0,
+      // help TS infer aggregations by making all aggregations required
+      aggs: filterAggregations as Required<typeof filterAggregations>
+    }
+  });
+
+  const response = await client.search(params);
+
+  return localFilterNames.map(key => {
+    const aggregations = response.aggregations[key];
+    const filter = localUIFilters[key];
+
+    return {
+      ...filter,
+      options: sortByOrder(
+        aggregations.by_terms.buckets.map(bucket => {
+          return {
+            name: bucket.key,
+            count:
+              'bucket_count' in bucket
+                ? bucket.bucket_count.value
+                : bucket.doc_count
```

Comment:
Good point. I think today it's true that `showCount` is only false when there is no `bucket_count` (ie no terms aggregation). This is just for metrics. But I think in principle, `showCount` is a UI property. In the future there might be projections that are not aggregated, but we still want to show the count. Even today, I think a `showCount` UI prop is more clear than detecting whether we have a bucket count or not and displaying or hiding the count based on that.

Reply:
> In the future there might be projections that are not aggregated, but we still want to show the count

I think adding features for potential future gain could be problematic. We are adding cruft that currently doesn't add any value and increases maintenance. It's definitely not a lot in this case but I think simplifying an API if possible is always preferable.

> Even today, I think a showCount UI prop is more clear than detecting whether we have a bucket count or not and displaying or hiding the count based on that.

I think adding the following to the Filter component would be even more clear:
```js
const showCount = option.count != null;
```

If someone who is a little less familiar with the code base set `<Filter showCount={true} ...>` for metrics they will still not see a count (if you end up removing `bucket.doc_count` as suggested above).
If you don't remove `bucket.doc_count` they'll see that as the count which is still probably not what they wanted. Worst case they see the `doc_count` and think that it works without giving it more thought. The fact is: if we don't have `count` it's because we don't have a count to show.

Reply:
I'm not sure "this doesn't have a count because we don't aggregate on terms" is a good explanation to someone who is new to the codebase either. Will change, though.

Reply:
As I'm implementing this, I'm running into an issue with the transaction group detail page. E.g. should we show a count here:

![image](https://user-images.githubusercontent.com/352732/62859397-4a408180-bcfd-11e9-8ea3-6d61eab8acbd.png)

Probably not. But maybe the issue is with the projection, because we don't care about aggregations on this page (ie, we're looking at transactions, and not transaction groups). What do you think, keep as-is, re-add `showCount` or create a different projection?

Reply:
I think it make sense to use `doc_count` as count here. Same with the error group details page.